src/cider/nrepl/middleware/util/instrument.clj

```diff
@@ -99,8 +99,13 @@
             '#{.} `(~(instrument (first args))
                     ~(second args)
                     ~@(instrument-coll (rest (rest args))))
-            '#{def set!} (list* (with-meta-safe (first args) {:cider-instrumented true})
-                                (map instrument (rest args)))
+            '#{def} (let [sym (first args)]
+                      (list* (if (meta sym)
+                               (with-meta-safe sym (instrument (meta sym)))
```

Comment:
Also assoc `:cider-instrumented true` in the return value of `(instrument (meta sym))`.